### ISSUE TITLE
E3002 - ECR Repository ScanOnPush Property Failing Validation 

### ISSUE THREAD

**Issue by andmarek** [aws-cloudformation/cfn-lint#2081]
*cfn-lint version: (`cfn-lint --version`)* 0.52.0

*Description of issue.*

I am creating an ECR Repository using the AWS CDK in Python.  After synthesizing, I am left with something like this:
```
"my-stack": {
      "Type": "AWS::ECR::Repository",
      "Properties": {
        "ImageScanningConfiguration": {
          "ScanOnPush": true
        },
        "RepositoryName": "my-repo",
```
In my pipeline, I am running cfn-lint 0.52.0 and receive this:
```
E3002 Invalid Property Resources/<resourceid>/Properties/ImageScanningConfiguration/scanOnPush
cdk.out/example-env/example-region/example-template-name.template.json:7:11
```

I am able to deploy this template just fine using the ```cdk deploy``` command.

This leads me to believe that cfn-lint incorrectly errors on ImageScanOnPush property for ECR repositories.
ImageScanOnPush is a supported property for CFN, specification and documentation needs to be updated.

[Here is the documentation for the AWS CDK that specifies that this should be supported.](https://docs.aws.amazon.com/cdk/api/latest/python/aws_cdk.aws_ecr/README.html), and the [CloudFormation Docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecr-repository-imagescanningconfiguration.html).

Cfn-lint uses the [CloudFormation Resource Specifications](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) as the base to do validation. These files are included as part of the application version. Please update to the latest version of `cfn-lint` or update the spec files manually (`cfn-lint -u`)

**Comment by kddejong** [comment]
Something doesn't match from your error to your template. Your example has `ScanOnPush` and the error has `scanOnPush`  The property for sure exists in the resource spec as this template does not have an issue.  Are you sure that's the right resource definition?
```
Resources:
  repo:
    Type: "AWS::ECR::Repository"
    Properties:
      ImageScanningConfiguration:
        ScanOnPush: true
      RepositoryName: "my-repo"
```

**Comment by andmarek** [comment]
I would imagine it's the correct resource definition, as I am generating it using the AWS CDK in Python.  Perhaps it's something to take up over there?  Just for clarity, I am writing this in CDK code:

```python
ecr.Repository(
            self,
            id="example-id",
            image_scan_on_push=True,
            repository_name="example-repo-name"
        )
```

It is generating a template similar to this:
```json
"Resources": {
    "example-id": {
      "Type": "AWS::ECR::Repository",
      "Properties": {
        "ImageScanningConfiguration": {
          "ScanOnPush": true
        },
        "RepositoryName": "example-repo-name",
      },
      "UpdateReplacePolicy": "Retain",
      "DeletionPolicy": "Retain",
      "Metadata": {
        "aws:cdk:path": "example/example/Resource"
      }
    }
```
I am running:
```
cfn-lint -t .\readss-dash-app-aws-pgi.template.json
```

And receiving:
```
E3002 Invalid Property Resources/example-id/Properties/ImageScanningConfiguration
.\example-name.template.json:6:9
```

The error that had the *scanOnPush* was something the pipeline spit out.  Running it locally, it actually didn't specify that.  The pipeline uses the latest version of cfn-lint, as noted previously.

**Comment by kddejong** [comment]
Not sure what package you are using but we both use the CloudFormation spec.   I would need to look back and see if there was a change that caused that.

**Comment by kddejong** [comment]
https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/245
https://github.com/aws/aws-cdk/issues/13418
https://github.com/aws/aws-cdk/pull/13420/files
